pkg/volumes/do/volumes.go

```diff
@@ -136,54 +155,136 @@ func (a *DOVolumes) findVolumes(filterByRegion bool) ([]*volumes.Volume, error)
+		glog.V(2).Infof("Iterating DO Volume with name=%s ID=%s, nametag=%s", doVolume.Name, doVolume.ID, a.nameTag)
 
-		// determine if this volume belongs to this cluster
-		// check for string a.ClusterName but with strings "." replaced with "-"
-		// example volume name will be like kops-1-etcd-events-dev5-techthreads-co-in
-		var k8sclusterName = a.matchTags["kubernetes.io/cluster"]
-
-		if !strings.Contains(doVolume.Name, strings.Replace(k8sclusterName, ".", "-", -1)) {
-			glog.V(2).Infof("DO Volume with name=%s does not match our cluster name=%s", doVolume.Name, k8sclusterName)
-			continue
+		for _, volumeTag := range doVolume.Tags {
+			glog.V(2).Infof("Iterating all Volume tags for volume name=%s volumetag=%s", doVolume.Name, volumeTag)
 		}
 
-		// Todo - Utilize volume tags once the change is done on the KOPS side. This still works since we specify a unique
-		// volume name when creating a volume via kOPS.
-		var clusterKey string
-		if strings.Contains(doVolume.Name, "etcd-main") {
-			clusterKey = "main"
-		} else if strings.Contains(doVolume.Name, "etcd-events") {
-			clusterKey = "events"
-		} else {
-			clusterKey = ""
-			glog.V(2).Infof("could not determine etcd cluster type for volume: %s", doVolume.Name)
+		for _, dropletTag := range a.dropletTags {
+			glog.V(2).Infof("Iterating all droplet tags for droplet name=%s dropletTag=%s", a.dropletName, dropletTag)
 		}
 
-		if strings.Contains(a.nameTag, clusterKey) {
-			glog.V(2).Infof("Found a matching nameTag=%s for cluster key=%s with etcd cluster name = %s and k8s cluster name=%s", a.nameTag, clusterKey, a.ClusterName, k8sclusterName)
-			vol := &volumes.Volume{
-				ProviderID: doVolume.ID,
-				Info: volumes.VolumeInfo{
-					Description: a.ClusterName + "-" + a.nameTag,
-				},
-				MountName: "master-" + doVolume.ID,
-				EtcdName:  a.ClusterName + "-" + clusterKey,
+		// make sure dropletTags match the volumeTags for the keys mentioned in matchTags
+		tagFound := a.matchesTags(&doVolume)
+
+		if tagFound {
+
+			glog.V(2).Infof("Tag Matched for droplet name=%s and volume name=%s", a.dropletName, doVolume.Name)
+
+			var clusterKey string
+			if strings.Contains(doVolume.Name, "etcd-main") {
+				clusterKey = "main"
+			} else if strings.Contains(doVolume.Name, "etcd-events") {
+				clusterKey = "events"
+			} else {
+				clusterKey = ""
+				glog.V(2).Infof("could not determine etcd cluster type for volume: %s", doVolume.Name)
 			}
 
-			if len(doVolume.DropletIDs) == 1 {
-				vol.AttachedTo = strconv.Itoa(doVolume.DropletIDs[0])
-				vol.LocalDevice = getLocalDeviceName(&doVolume)
+			if strings.Contains(a.nameTag, clusterKey) {
+				vol := &volumes.Volume{
+					ProviderID: doVolume.ID,
+					Info: volumes.VolumeInfo{
+						Description: a.ClusterName + "-" + a.nameTag,
+					},
+					MountName: "master-" + doVolume.ID,
+					EtcdName:  doVolume.Name,
+				}
+
+				if len(doVolume.DropletIDs) == 1 {
```

Comment:
Nit / tactic: it's probably a good idea to say `> 0` just in case DO ever supports multiple attachment.  You can then return an error if `len() > 1` (e.g. `if len(...) > 1 { return fmt.Errorf("multiple attachments found for volume %q", doVolume.Name) }`)

Reply:
Sure, updated.